File: 0169-majority-element/0169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let majorityElem = null;
    let count = 0;
    for(const num of nums){
        if(count === 0){
            majorityElem = num;
        }

        if(num === majorityElem) count++;
        else count--;
    }
    return majorityElem;
};

File: 0169-majority-element/0169-majority-element.cs
public class Solution {
    public int MajorityElement(int[] nums) {
        // Dictionary to get the count for each element.
        Dictionary<int, int> elements = new Dictionary<int, int>();
        int winner = nums[0];
        int count = 0;
        for(var i = 0; i < nums.Length; i++){
           if(count == 0){
            winner = nums[i];
           }
           if(winner == nums[i]) count++;
           else count--;
        }
        return winner;
    }
}